draft-ietf-jsonpath-base.md

```diff
@@ -577,207 +590,227 @@ of node.
-                 union-element ws "," ws union-elements
-                                       ; ,-separated list
+dot-selector    = "." dot-member-name
+dot-member-name = 1*(
+                      DIGIT /
+                      ALPHA /
+                      "_"   /           ; _
+                      %x80-10FFFF       ; any non-ASCII Unicode character
+                    )
+DIGIT           =  %x30-39              ; 0-9
+ALPHA           =  %x41-5A / %x61-7A    ; A-Z / a-z
 ~~~~
 
+Member names containing other ASCII characters than allowed by `dot-selector` &ndash; especially space `' '` and minus `'-'` characters MUST NOT be used with the `dot-selector`. Member names containing these might be addressed by the `index-selector` instead.
 
 #### Semantics
+{: numbered="false" toc="exclude"}
 
-A union selects any node which is selected by at least one of the union selectors
-and selects the concatenation of the
-lists (in the order of the selectors) of nodes selected by the union elements.<!--  TODO: define whether duplicates are kept or removed.  -->
+The `dot-selector` selects the value corresponding to the member name from any JSON object. It selects no values from any other JSON value.
 
+Note that the `dot-selector` follows the philosophy of JSON strings and is
+allowed to contain bit sequences that cannot encode Unicode characters (a
+single unpaired UTF-16 surrogate, for example).
+The behaviour of an implementation is undefined for member names which do
+not encode Unicode characters.
 
-#### Child {#unionchild}
+### Dot Wild Card Selector
 
-##### Syntax
+#### Syntax
 {: numbered="false" toc="exclude"}
 
-A child is a quoted string.
+The dot wild card selector has the shape `'.*'`.
 
 ~~~~ abnf
-union-element = child ; see below for more alternatives
-child = %x22 *double-quoted %x22 / ; "string"
-        "'" *single-quoted "'"   ; 'string'
+dot-wild-selector    = "." "*"            ;  dot followed by asterisk
+~~~~
+
+#### Semantics
+{: numbered="false" toc="exclude"}
+
+A dot selector combined with a single asterisk is a wild card. It selects all member values of an object as well as all elements of an array. Applying the `dot-wild-selector` to primitive JSON values, as number, string, or literal, does nothing.
 
-double-quoted = dq-unescaped /
-          escape (
-              %x22 /         ; "    quotation mark  U+0022
-              "/" /          ; /    solidus         U+002F
-              "\" /          ; \    reverse solidus U+005C
-              "b" /          ; b    backspace       U+0008
-              "f" /          ; f    form feed       U+000C
-              "n" /          ; n    line feed       U+000A
-              "r" /          ; r    carriage return U+000D
-              "t" /          ; t    tab             U+0009
-              "u" 4HEXDIG )  ; uXXXX                U+XXXX
 
+### Index Selector
 
-dq-unescaped = %x20-21 / %x23-5B / %x5D-10FFFF
+#### Syntax
+{: numbered="false" toc="exclude"}
```

Comment:
If we want to use unnumbered sections more often, we might want to make an ALD for this to reduce noise.
Unnumbered sections are a bit of a boobytrap as they have to be used consistently in the XML.

Reply:
Done in #102